ocaml/typing/printtyped.ml

```diff
@@ -1037,10 +1037,17 @@ and case
   pattern (i+1) ppf c_lhs;
   begin match c_guard with
   | None -> ()
-  | Some g -> line (i+1) ppf "<when>\n"; expression (i + 2) ppf g
+  | Some g -> line (i+1) ppf "<when>\n"; guard (i + 2) ppf g
   end;
   expression (i+1) ppf c_rhs;
 
+and guard i ppf = function
+  | Predicate p -> expression i ppf p
+  | Pattern (e, s, pat) ->
+      expression i ppf e;
+      line i ppf "%a\n" Layouts.Sort.format s;
+      list i pattern ppf [ pat ]
```

Comment:
I think we could change this to just `pattern i ppf pat`. (That is, I don't see why we need to print this as though a list.)

Reply:
Sure, done!